pkg/docker/docker.go

```diff
@@ -608,70 +606,81 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a SIGINT/SIGTERM/SIGQUIT/SIGHUP.
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
-	}()
-
-	// Attach to the container.
-	glog.V(2).Infof("Attaching to container %q ...", containerName)
-	attachOpts := opts.asDockerAttachToContainerOptions()
-	attachOpts.Container = container.ID
-	if _, err = d.client.AttachToContainerNonBlocking(attachOpts); err != nil {
-		glog.Errorf("Unable to attach to container %q with options %+v: %v", containerName, attachOpts, err)
-		return err
-	}
-
-	// Start the container.
-	glog.V(2).Infof("Starting container %q ...", containerName)
-	if err := util.TimeoutAfter(DefaultDockerTimeout, "timeout after waiting %v for Docker to start container", func() error {
-		return d.client.StartContainer(container.ID, nil)
-	}); err != nil {
-		return err
+		removalDone <- struct{}{}
 	}
+	dumpStack := func(signal os.Signal) {
+		if signal == syscall.SIGQUIT {
+			buf := make([]byte, 1<<16)
```

Comment:
I wonder how the runtime handles this... All times I've seen this piece of dump stack code in the recent past, I'm wondering what's the reasonable buffer size we need to allocate. We don't want it to be too big, while too small would truncate the dump?!